From fed869414aa22aeea1c6e971a0600df3d5d0077e Mon Sep 17 00:00:00 2001
From: Frank Hunleth <fhunleth@troodon-software.com>
Date: Thu, 21 Mar 2019 20:49:54 -0400
Subject: [PATCH] erlang: enable deterministic builds

This adds the `+deterministic` compiler flag to the `erlc` calls to
strip out absolute paths in compiled `.beam` files.

Signed-off-by: Frank Hunleth <fhunleth@troodon-software.com>
---
 make/otp.mk.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/make/otp.mk.in b/make/otp.mk.in
index 63748c2f2b..c31cbfe971 100644
--- a/make/otp.mk.in
+++ b/make/otp.mk.in
@@ -114,6 +114,7 @@ ifeq ($(USE_ESOCK),yes)
 ERL_COMPILE_FLAGS += -DUSE_ESOCK=true
 endif
 
+ERL_COMPILE_FLAGS += +deterministic
 ERLC_WFLAGS = -W
 ERLC = erlc $(ERLC_WFLAGS) $(ERLC_FLAGS)
 ERL = erl -boot start_clean
-- 
2.17.1

